[v2.7.0 broken; new version needed] Tests do not even compile

I am responsible for https://aur.archlinux.org/packages/cli11-compiled which is broken now. The build instructions are [here](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=cli11-compiled). A new version is needed.

```sh
[ 40%] Linking CXX executable TransformTest
[ 40%] Built target TransformTest
[ 41%] Building CXX object tests/CMakeFiles/CreationTest.dir/CreationTest.cpp.o
[ 43%] Linking CXX executable CreationTest
/usr/bin/ld: /tmp/ccWgLCLI.ltrans0.ltrans.o: in function `(anonymous namespace)::CATCH2_INTERNAL_TEST_4::test() [clone .lto_priv.0]':
/usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:27:(.text+0x3df8): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /tmp/ccWgLCLI.ltrans0.ltrans.o: in function `(anonymous namespace)::CATCH2_INTERNAL_TEST_6::test() [clone .lto_priv.0]':
/usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:32:(.text+0x3fb8): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /tmp/ccWgLCLI.ltrans0.ltrans.o: in function `(anonymous namespace)::CATCH2_INTERNAL_TEST_8::test() [clone .lto_priv.0]':
/usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:37:(.text+0x4178): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /tmp/ccWgLCLI.ltrans0.ltrans.o: in function `(anonymous namespace)::CATCH2_INTERNAL_TEST_16::test() [clone .lto_priv.0]':
/usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:60:(.text+0x48c8): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /tmp/ccWgLCLI.ltrans0.ltrans.o: in function `(anonymous namespace)::CATCH2_INTERNAL_TEST_18::test() [clone .lto_priv.0]':
/usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:67:(.text+0x4a96): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /tmp/ccWgLCLI.ltrans0.ltrans.o: in function `(anonymous namespace)::CATCH2_INTERNAL_TEST_20::test() [clone .lto_priv.0]':
/usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:74:(.text+0x4c74): undefined reference to `CLI::Option* CLI::Option::ignore_underscore<CLI::App>(bool)'
/usr/bin/ld: /tmp/ccWgLCLI.ltrans0.ltrans.o: in function `(anonymous namespace)::CATCH2_INTERNAL_TEST_22::test() [clone .lto_priv.0]':
/usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:81:(.text+0x4e54): undefined reference to `CLI::Option* CLI::Option::ignore_underscore<CLI::App>(bool)'
/usr/bin/ld: /tmp/ccWgLCLI.ltrans0.ltrans.o: in function `(anonymous namespace)::CATCH2_INTERNAL_TEST_82::test() [clone .lto_priv.0]':
/usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:300:(.text+0x6a23): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:301:(.text+0x6a6f): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:302:(.text+0x6ab7): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:303:(.text+0x6b03): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:305:(.text+0x6b7e): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /tmp/ccWgLCLI.ltrans0.ltrans.o:/usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:306: more undefined references to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)' follow
/usr/bin/ld: /tmp/ccWgLCLI.ltrans1.ltrans.o: in function `(anonymous namespace)::CATCH2_INTERNAL_TEST_84::test() [clone .lto_priv.0]':
/usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:328:(.text+0xa93): undefined reference to `CLI::Option* CLI::Option::ignore_underscore<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:329:(.text+0xadf): undefined reference to `CLI::Option* CLI::Option::ignore_underscore<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:332:(.text+0xb56): undefined reference to `CLI::Option* CLI::Option::ignore_underscore<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:333:(.text+0xbd4): undefined reference to `CLI::Option* CLI::Option::ignore_underscore<CLI::App>(bool)'
/usr/bin/ld: /tmp/ccWgLCLI.ltrans1.ltrans.o: in function `(anonymous namespace)::CATCH2_INTERNAL_TEST_86::test() [clone .lto_priv.0]':
/usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:359:(.text+0x1f23): undefined reference to `CLI::Option* CLI::Option::ignore_underscore<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:359:(.text+0x1f31): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:360:(.text+0x1f7d): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:360:(.text+0x1f8b): undefined reference to `CLI::Option* CLI::Option::ignore_underscore<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:363:(.text+0x2002): undefined reference to `CLI::Option* CLI::Option::ignore_underscore<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:363:(.text+0x2010): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:364:(.text+0x208e): undefined reference to `CLI::Option* CLI::Option::ignore_case<CLI::App>(bool)'
/usr/bin/ld: /usr/src/debug/cli11-compiled/CLI11-2.7.0/tests/CreationTest.cpp:364:(.text+0x209c): undefined reference to `CLI::Option* CLI::Option::ignore_underscore<CLI::App>(bool)'
collect2: error: ld returned 1 exit status
make[2]: *** [tests/CMakeFiles/CreationTest.dir/build.make:104: tests/CreationTest] Error 1
make[1]: *** [CMakeFiles/Makefile2:1620: tests/CMakeFiles/CreationTest.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
==> ERROR: A failure occurred in check().
    Aborting...
error: failed to build 'cli11-compiled-2.7.0-1':
error: packages failed to build: cli11-compiled-2.7.0-1
```

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -82,6 +82,11 @@ jobs:
             options: -DCMAKE_CXX_FLAGS="-std=c++23 -stdlib=libc++"
               -DCLI11_ENABLE_EXTRA_VALIDATORS=1 -DCMAKE_CXX_COMPILER=clang++
               -DCLI11_FORCE_LIBCXX=ON -DCMAKE_LINKER_TYPE=LLD
+          - name: gcc15 std17 precompiled shared lto
+            container: gcc:15
+            std: 17
+            precompile: ON
+            options: -DBUILD_SHARED_LIBS=ON -DCMAKE_CXX_FLAGS="-O2 -flto=auto"
           - name: clang20 std26
             container: silkeh/clang:20
             std: 26

--- a/src/Precompile.cpp
+++ b/src/Precompile.cpp
@@ -4,6 +4,13 @@
 //
 // SPDX-License-Identifier: BSD-3-Clause
 
+// Build the full validator set into the library by default so consumers that
+// define CLI11_ENABLE_EXTRA_VALIDATORS per-TU can still link. An explicit
+// CMake-level choice (passed as a PUBLIC definition) is respected.
+#if !defined(CLI11_ENABLE_EXTRA_VALIDATORS) && !defined(CLI11_DISABLE_EXTRA_VALIDATORS)
+#define CLI11_ENABLE_EXTRA_VALIDATORS 1
+#endif
+
 // IWYU pragma: begin_keep
 
 #include <CLI/impl/App_inl.hpp>
@@ -18,3 +25,12 @@
 #include <CLI/impl/Validators_inl.hpp>
 
 // IWYU pragma: end_keep
+
+namespace CLI {
+// Library consumers only see the declarations of these member templates, so
+// the compiled library must provide the instantiations. Implicit instantiation
+// emits only discardable weak symbols, which LTO can remove from a shared
+// library; instantiate explicitly so the symbols always survive.
+template Option *Option::ignore_case<App>(bool);
+template Option *Option::ignore_underscore<App>(bool);
+}  // namespace CLI

